src/image-tvg.c

```diff
@@ -0,0 +1,1453 @@
+        if (res != TVG_SUCCESS) {
+            return res;
+        }
+        out_style->flat = flat;
+        break;
+    case TVG_STYLE_LINEAR:
+        res = tvg_parse_gradient(ctx, &grad);
+        out_style->linear = grad;
+        break;
+    case TVG_STYLE_RADIAL:
+        res = tvg_parse_gradient(ctx, &grad);
+        out_style->radial = grad;
+        break;
+    default:
+        res = TVG_E_INVALID_FORMAT;
+        break;
+    }
+    return res;
+}
+
+static tvg_result_t tvg_parse_fill_header(tvg_context_t *ctx,
+                                          int kind,
+                                          tvg_fill_header_t *out_header)
+{
+    uint32_t u32;
+    tvg_result_t res = tvg_read_varuint(ctx, &u32);
+    if (res != TVG_SUCCESS) {
+        return res;
+    }
+    size_t count = (size_t) u32 + 1;
+    out_header->size = count;
+    res = tvg_parse_style(ctx, kind, &out_header->style);
+    if (res != TVG_SUCCESS) {
+        log_error("Failed to parse style, error = [%d]", res);
+    }
+    return res;
+}
+
+static tvg_result_t tvg_parse_line_header(tvg_context_t *ctx,
+                                          int kind,
+                                          tvg_line_header_t *out_header)
+{
+    uint32_t u32;
+    tvg_result_t res = tvg_read_varuint(ctx, &u32);
+    if (res != TVG_SUCCESS) {
+        return res;
+    }
+    size_t count = (size_t) u32 + 1;
+    out_header->size = count;
+    res = tvg_parse_style(ctx, kind, &out_header->style);
+    if (res != TVG_SUCCESS) {
+        log_error("Failed to parse style, error = [%d]", res);
+        return res;
+    }
+    res = tvg_read_unit(ctx, &out_header->line_width);
+    if (res != TVG_SUCCESS) {
+        log_error("Failed to read unit, error = [%d]", res);
+        return res;
+    }
+
+    return TVG_SUCCESS;
+}
+
+static tvg_result_t tvg_parse_line_fill_header(
+    tvg_context_t *ctx,
+    int kind,
+    tvg_line_fill_header_t *out_header)
+{
+    uint8_t d;
+    if (1 > ctx->in(&d, 1, ctx->inp_state)) {
+        return TVG_E_IO_ERROR;
+    }
+    tvg_result_t res = TVG_SUCCESS;
+
+    size_t count = TVG_SIZE_AND_STYLE_SIZE(d);
+    out_header->size = count;
+    res = tvg_parse_style(ctx, kind, &out_header->fill_style);
+    if (res != TVG_SUCCESS) {
+        log_error("Failed to parse style, error = [%d]", res);
+        return res;
+    }
+    res = tvg_parse_style(ctx, TVG_SIZE_AND_STYLE_STYLE_KIND(d),
+                          &out_header->line_style);
+    if (res != TVG_SUCCESS) {
+        log_error("Failed to parse style, error = [%d]", res);
+        return res;
+    }
+    res = tvg_read_unit(ctx, &out_header->line_width);
+    if (res != TVG_SUCCESS) {
+        log_error("Failed to read unit, error = [%d]", res);
+        return res;
+    }
+
+    return TVG_SUCCESS;
+}
+
+static tvg_result_t tvg_parse_path(tvg_context_t *ctx, size_t size)
+{
+    tvg_result_t res = TVG_SUCCESS;
+    tvg_point_t st, cur, pt;
+    float f32;
+    uint8_t d;
```

Comment:
`d` Could it be given a different variable name? The current name is hard to understand.